Incorrect error when missing repomap.csv or not being on 7.6

When `/etc/leapp/files/repomap.csv` is missing, the upgrade is not inhibited by the _repositoriesmapping_ actor with the message ["_Repositories map file not found_"](https://github.com/oamg/leapp-repository/blob/master/repos/system_upgrade/el7toel8/actors/repositoriesmapping/libraries/library.py#L20).
Instead I'm getting:
```
2019-08-09 17:08:03.78  DEBUG    PID: 28670 leapp.workflow.FactsCollection.target_userspace_creator: Target RHSM Info: SKUs: [u'ES0113909'] Repositories: [u'rhel-8-for-x86_64-baseos-rpms', u'rhel-8-for-x86_64-appstream-rpms']
2019-08-09 17:08:03.430 DEBUG    PID: 28670 leapp.workflow.FactsCollection.target_userspace_creator: Gathered target repositories:
...
2019-08-09 17:08:27.251042 [ERROR] Actor: target_userspace_creator Message: There are no enabled target repositories for the upgrade process to proceed.
Detail: {u'hint': u'Ensure your system is correctly registered with the subscription manager and that your current subscription is entitled to install the requested target version 8.0'}
```

The _repositoriesmapping_ needs to run before the _targetuserspacecreator_. @vinzenz, don't you know of a better solution than sending a message between these just to create a dependency link?

A new but very similar issue: https://access.redhat.com/discussions/4356841

The [CheckOSRelease](https://github.com/oamg/leapp-repository/pull/247) actor that should allow upgrade of RHEL 7.6 only is being executed too late. 

Instead of a clear message that only 7.6 is supported, customers get:
```
2019-08-19 15:00:24.363331 [ERROR] Actor: target_userspace_creator Message: A subscription-manager command failed to execute
Detail: {u'hint': u'Please ensure you have a valid RHEL subscription and your network is up.'}
```

--- a/repos/system_upgrade/el7toel8/actors/checkosrelease/actor.py
+++ b/repos/system_upgrade/el7toel8/actors/checkosrelease/actor.py
@@ -1,5 +1,6 @@
 from leapp.actors import Actor
 from leapp.libraries.actor.library import check_os_version, skip_check
+from leapp.libraries.common.config import version
 from leapp.models import OSReleaseFacts
 from leapp.reporting import Report
 from leapp.tags import ChecksPhaseTag, IPUWorkflowTag
@@ -20,6 +21,5 @@ class CheckOSRelease(Actor):
     tags = (ChecksPhaseTag, IPUWorkflowTag)
 
     def process(self):
-        supported_version = {'rhel': ['7.6']}
         if not skip_check():
-            check_os_version(supported_version)
+            check_os_version(version.SUPPORTED_VERSION)

--- a/repos/system_upgrade/el7toel8/actors/checkosrelease/libraries/library.py
+++ b/repos/system_upgrade/el7toel8/actors/checkosrelease/libraries/library.py
@@ -3,7 +3,7 @@ import os
 from leapp.exceptions import StopActorExecution, StopActorExecutionError
 from leapp import reporting
 from leapp.libraries.stdlib import api
-from leapp.models import OSReleaseFacts
+from leapp.libraries.common.config import version
 
 
 COMMON_REPORT_TAGS = [reporting.Tags.SANITY]
@@ -29,16 +29,9 @@ def check_os_version(supported_version):
         api.current_logger().warning('The supported version value is invalid.')
         raise StopActorExecution()
 
-    facts_messages = api.consume(OSReleaseFacts)
-    facts = next(facts_messages, None)
-    if list(facts_messages):
-        api.current_logger().warning('Unexpectedly received more than one OSReleaseFacts message.')
-    if not facts:
-        raise StopActorExecutionError(
-            'Could not check OS version', details={'details': 'No OSReleaseFacts facts found.'}
-        )
+    release_id, version_id = version.current_version()
 
-    if facts.release_id not in supported_version:
+    if not version.matches_release(supported_version.keys(), release_id):
         reporting.create_report([
             reporting.Title('Unsupported OS'),
             reporting.Summary('Only RHEL is supported by the upgrade process'),
@@ -49,18 +42,18 @@ def check_os_version(supported_version):
 
         return
 
-    if not isinstance(supported_version[facts.release_id], list):
+    if not isinstance(supported_version[release_id], list):
         raise StopActorExecutionError(
             'Invalid versions',
             details={'details': 'OS versions are invalid, please provide a valid list.'},
         )
 
-    if facts.version_id not in supported_version[facts.release_id]:
+    if not version.matches_version(supported_version[release_id], version_id):
         reporting.create_report([
             reporting.Title('Unsupported OS version'),
             reporting.Summary(
                 'The supported OS versions for the upgrade process: {}'.format(
-                    ', '.join(supported_version[facts.release_id])
+                    ', '.join(supported_version[release_id])
                 )
             ),
             reporting.Severity(reporting.Severity.HIGH),

--- a/repos/system_upgrade/el7toel8/actors/dnftransactioncheck/actor.py
+++ b/repos/system_upgrade/el7toel8/actors/dnftransactioncheck/actor.py
@@ -22,5 +22,6 @@ class DnfTransactionCheck(Actor):
         tasks = next(self.consume(FilteredRpmTransactionTasks), FilteredRpmTransactionTasks())
         target_userspace_info = next(self.consume(TargetUserSpaceInfo), None)
 
-        dnfplugin.perform_transaction_check(tasks=tasks, used_repos=used_repos,
-                                            target_userspace_info=target_userspace_info, xfs_present=xfs_present)
+        if target_userspace_info:
+            dnfplugin.perform_transaction_check(tasks=tasks, used_repos=used_repos,
+                                                target_userspace_info=target_userspace_info, xfs_present=xfs_present)

--- a/repos/system_upgrade/el7toel8/actors/targetuserspacecreator/actor.py
+++ b/repos/system_upgrade/el7toel8/actors/targetuserspacecreator/actor.py
@@ -1,7 +1,8 @@
 from leapp.actors import Actor
 from leapp.libraries.actor import userspacegen
-from leapp.models import (RequiredTargetUserspacePackages, SourceRHSMInfo, TargetRepositories,
-                          TargetRHSMInfo, TargetUserSpaceInfo, UsedTargetRepositories, XFSPresence)
+from leapp.libraries.common.config import version
+from leapp.models import (OSReleaseFacts, RepositoriesMap, RequiredTargetUserspacePackages, SourceRHSMInfo,
+                          TargetRepositories, TargetRHSMInfo, TargetUserSpaceInfo, UsedTargetRepositories, XFSPresence)
 from leapp.tags import FactsPhaseTag, IPUWorkflowTag
 
 
@@ -17,9 +18,11 @@ class TargetUserspaceCreator(Actor):
     """
 
     name = 'target_userspace_creator'
-    consumes = (RequiredTargetUserspacePackages, SourceRHSMInfo, TargetRepositories, XFSPresence)
+    consumes = (OSReleaseFacts, RepositoriesMap, RequiredTargetUserspacePackages,
+                SourceRHSMInfo, TargetRepositories, XFSPresence)
     produces = (TargetRHSMInfo, TargetUserSpaceInfo, UsedTargetRepositories)
     tags = (IPUWorkflowTag, FactsPhaseTag)
 
     def process(self):
-        userspacegen.perform()
+        if version.is_supported_version() and next(self.consume(RepositoriesMap), None):
+            userspacegen.perform()

--- a/repos/system_upgrade/el7toel8/libraries/config/version.py
+++ b/repos/system_upgrade/el7toel8/libraries/config/version.py
@@ -1,9 +1,11 @@
 import operator
+import os
 
 import six
 
+from leapp.exceptions import StopActorExecutionError
 from leapp.libraries.stdlib import api
-
+from leapp.models import OSReleaseFacts
 
 OP_MAP = {
     '>': operator.gt,
@@ -11,6 +13,8 @@ OP_MAP = {
     '<': operator.lt,
     '<=': operator.le
 }
+
+SUPPORTED_VERSION = {'rhel': ['7.6']}
 
 
 def _version_to_tuple(version):
@@ -109,3 +113,54 @@ def matches_target_version(*match_list):
     """
     target_version = api.current_actor().configuration.version.target
     return matches_version(match_list, target_version)
+
+
+def matches_release(allowed_releases, release):
+    """Check if the given `release` is allowed to upgrade based in `allowed_releases`.
+
+    :param allowed_releases: All supported releases
+    :type allowed_releases: list or dict
+    :param release: release name to be checked
+    :type release: string
+    :return: ``True`` if `release` value matches one of the values in `allowed_releases`, ``False`` otherwise
+    :rtype: bool
+    """
+    if not (release and allowed_releases):
+        return False
+
+    return release in allowed_releases
+
+
+def current_version():
+    """Return current Linux release and version running in system.
+
+    :raises StopActorExecutionError: this exception will be raised if no facts found to read data.
+    :return: The tuple contains release name and version value.
+    :rtype: (string, string)
+    """
+    facts_messages = api.consume(OSReleaseFacts)
+    facts = next(facts_messages, None)
+    if list(facts_messages):
+        api.current_logger().warning('Unexpectedly received more than one OSReleaseFacts message.')
+    if not facts:
+        raise StopActorExecutionError(
+            'Could not check OS version', details={'details': 'No OSReleaseFacts facts found.'}
+        )
+    return facts.release_id, facts.version_id
+
+
+def is_supported_version():
+    """ Verify if current system version is supported to the upgrade.
+
+    :return: `True` if current version is supported and `False` otherwise.
+    :rtype: bool
+    """
+    if os.getenv('LEAPP_SKIP_CHECK_OS_RELEASE'):
+        return True
+
+    release_id, version_id = current_version()
+
+    if not matches_release(SUPPORTED_VERSION, release_id):
+        return False
+
+    return matches_version(SUPPORTED_VERSION[release_id], version_id)

